ipalib/util.py

```diff
@@ -1078,8 +1077,8 @@ def check_client_configuration():
     """
     Check if IPA client is configured on the system.
     """
-    fstore = sysrestore.FileStore(paths.IPA_CLIENT_SYSRESTORE)
-    if not fstore.has_files() and not os.path.exists(paths.IPA_DEFAULT_CONF):
+    if (not any(os.listdir(IPA_CLIENT_SYSRESTORE)) and not
```

Comment:
This block is hard to understand. Why ``any()``? Also ``os.listdir()`` raises a FileNotFound / OSError exception when the directory does not exist.

```
if (not os.path.isfile(IPA_DEFAULT_CONF) or
       not os.path.isdir(IPA_CLIENT_SYSRESTORE) or
       not os.listdir(IPA_CLIENT_SYSRESTORE)):
    raise ...
```

Reply:
I used any to avoid the two checks if it's a file or a dir, since, IIUC, it's not relevant to this case. 

I'll update the PR to handle with a possible exception when the dir does not exist.